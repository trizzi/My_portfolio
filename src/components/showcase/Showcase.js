import React from 'react';
import { Link } from 'react-scroll';
import Button from '../button/Button';
import SlideIn from '../../utilities/SlideIn';
import FadeIn from '../../utilities/FadeIn';
import Text from '../../assets/Text (1).svg';
import showcaseImg from '../../assets/Profile picture (1).svg';

const Showcase = () => {
	const emailAddress = 'patrickakhamiogu@gmail.com';

	const handleEmailClick = () => {
		window.location.href = `mailto:${emailAddress}`;
	};

	return (
		<div
			name='section1'
			className='flex'>
			<div className='flex flex-col md:flex-row justify-between  md:container md:mx-auto items-center py-44'>
				<div className='px-12'>
					<FadeIn>
						<img
							src={Text}
							alt='Intro'
						/>
					</FadeIn>
					<Link to='section3'>
						<Button
							text={'Explore'}
							className='mt-4 text-white'
							background
						/>
					</Link>
				</div>
				<SlideIn>
					<div className='sm:mx-10 md:my-20 '>
						<div className='hidden md:flex text-white bg-[#29292E] opacity-90 rounded-full py-4 md:pl-4 md:pr-16 w-fit hover:bg-black'>
							<img
								src={showcaseImg}
								alt='Images'
							/>
							<a
								href={`mailto:${emailAddress}`}
								onClick={handleEmailClick}
								className='pt-3 pl-2'>
								<p>patrickakhamiogu@gmail.com</p>
							</a>
						</div>
					</div>
				</SlideIn>
			</div>
		</div>
	);
};

export default Showcase;
