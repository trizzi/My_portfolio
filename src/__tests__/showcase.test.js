import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Showcase from '../components/showcase/Showcase';
import { ThemeContextProvider } from '../context/ThemeContext';

// Mock the IntersectionObserver
beforeEach(() => {
	window.IntersectionObserver = jest.fn(function () {
		this.observe = jest.fn();
		this.disconnect = jest.fn();
		this.unobserve = jest.fn();
	});
});

describe('Showcase component', () => {
	test('renders thumbnail', () => {
		const { container } = render(
			<ThemeContextProvider>
				<Showcase />
			</ThemeContextProvider>,
		);

		const thumbnail = screen.getByTestId('thumbnail');
		expect(thumbnail).toBeInTheDocument();

		expect(container).toMatchSnapshot();
	});

	test('triggers email link on email click', () => {
		render(
			<ThemeContextProvider>
				<Showcase />
			</ThemeContextProvider>,
		);

		const emailLink = screen.getByText('patrickakhamiogu@gmail.com');
		fireEvent.click(emailLink);

		// Add additional assertions here based on the expected behavior
	});
});
