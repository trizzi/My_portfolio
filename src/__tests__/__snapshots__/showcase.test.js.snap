// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Showcase component renders thumbnail 1`] = `
<div>
  <div
    class="flex"
    name="section1"
  >
    <div
      class="flex flex-col md:flex-row justify-between  md:container md:mx-auto items-center py-44"
    >
      <div
        class="px-12"
      >
        <span
          style="opacity: 0;"
        >
          <img
            alt="Intro"
            data-testid="thumbnail"
            src="Text.svg"
          />
        </span>
        <a>
          <button
            class="
			rounded-lg py-2 lg:py-2 px-5 whitespace-nowrap hover:bg-violet-400
        undefined undefined mt-5"
            style="background-color: rgb(153, 50, 204); color: rgb(255, 255, 255); border: 2px solid #9932cc; opacity: 0;"
            tabindex="0"
          >
            Explore
          </button>
        </a>
      </div>
      <div
        style="transform: translateX(-1000px) translateZ(0);"
      >
        <div
          class="sm:mx-10 md:my-20 "
        >
          <div
            class="hidden md:flex text-white bg-[#29292E] opacity-90 rounded-full py-4 md:pl-4 md:pr-16 w-fit hover:bg-black"
          >
            <img
              alt="Images"
              src="profile-picture.svg"
            />
            <a
              class="pt-3 pl-2"
              href="mailto:patrickakhamiogu@gmail.com"
            >
              <p>
                patrickakhamiogu@gmail.com
              </p>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
